cmd/bench.go

```diff
@@ -316,6 +323,25 @@ func benchE2E(ctx context.Context, args []string, params benchmarkCommandParams,
 		break
 	}
 
+	// Busy loop until server has truly come online to recover the bound port.
+	// We do this with exponential backoff for wait times, since the server
+	// typically comes online very quickly.
+	delay := time.Duration(10) * time.Microsecond
+	retries := 7 // Max of 11 seconds total wait time.
+	for i := 0; i < retries; i++ {
+		if len(rt.Addrs()) == 0 {
+			time.Sleep(delay)
+			delay *= 10 // Up the next wait interval by 10x.
+			continue
+		}
+		// We have an address to parse the port from.
+		port, err = strconv.Atoi(strings.Split(rt.Addrs()[0], ":")[1])
+		if err != nil {
+			return err
+		}
+		break
+	}
```

Comment:
Do we need to check if `len(rt.Addrs()) == 0` after the loop ends? Also we have a [lib](https://github.com/open-policy-agent/opa/blob/main/util/backoff.go) that implements exponential backoff.

Reply:
Interesting that we have a library for that!

That's a good thought on checking for the failure condition after the loop. I'll work that in.